extend ivy api with nansum and add tests

popraf opened the pull request:
Hello, added nansum function and tests to ivy backend - issue #5880

popraf commented:
Hello @YushaArif99, I would be grateful for review and feedback on this PR. Thanks

YushaArif99 commented on ivy/container/extensions.py:
```diff
@@ -1370,3 +1370,125 @@ def fmax(
+        With one :class:`ivy.Container` input:
+        >>> x = ivy.Container(a=ivy.array([[10, 7, 4], [3, 2, 1]]),\
+                b=ivy.array([[1, 4, 2], [ivy.nan, ivy.nan, 0]]))
+        >>> x.nansum(axis=0)
```
Statis methods are not callable as an instance. Rather they are called on the class itself.  In this case, it would be `ivy.Container.static_nansum(x)`

YushaArif99 reviewed (changes requested):
Hi @popraf 

The following tests seem to be failing.

- lint / check-formatting 
- test-numpy-style-docstrings
- test-extensions-ivy (torch)

Also, could you kindly resolve the merge conflict? Thanks!

popraf commented:
Hello @YushaArif99, already requested for a second review, fixed merge conflicts, changed formatting, example has been modified.

Regarding the test-extensions-ivy tests failed: it throws "Error: Process completed with exit code 1" by a "shell: /usr/bin/bash -e {0}" according to logs. I've found the same issue with those tests in #5707 pr. Seems like the workflow configuration file might contain bugs, but I am not sure. Shall we continue with merging?

YushaArif99 commented:
Hi @popraf 

Sorry for the late review. Was a little busy with work and so reviewing this PR slipped under my radar. All the changes seem fine to me. The tests are also passing for the `nansum()` function. I am now going to merge this PR. Thank you for your contribution to the Ivy codebase 😊
